staging/src/k8s.io/legacy-cloud-providers/azure/azure_cache.go

```diff
@@ -69,16 +90,12 @@ func (t *timedCache) getInternal(key string) (*cacheEntry, error) {
 	if err != nil {
 		return nil, err
 	}
-	if exists {
-		return entry.(*cacheEntry), nil
-	}
-
+	// lock here to ensure if entry doesn't exist, we add a new entry
+	// avoiding overwrites
 	t.lock.Lock()
 	defer t.lock.Unlock()
```

Comment:
these two lines of lock should be moved to L102 (before newEntry insertion).

Reply:
Done!